### PR TITLE
Update the HMR docs for React

### PR DESCRIPTION
**Pull request by justin808**
Taking over from https://github.com/rails/webpacker/pull/2353

### CODE REVIEW

**Comment by justin808** [comment]
@gauravtiwari @jakeNiemiec any feedback?

**Comment by clearyandzap** [comment]
@justin808 You're missing one step: 

```
{
    "plugins": [
        [
            "react-hot-loader/babel",
            {
            "safetyNet": false
            }
        ]
    ]
}
```

Without this, it doesn't work.

**Comment by justin808** [comment]
@jakeNiemiec and @clearyandzap I'll get to your comments by this weekend.

@clearyandzap do you have any doc link for safetyNet item?

**Comment by clearyandzap** [comment]
@justin808 Yup, https://github.com/gaearon/react-hot-loader#getting-started

**Comment by justin808** [comment]
@clearyandzap docs there say:

> You **may** disable it to get more control on the module execution order.

Please clarify on why this is necessary with rails/webpacker.

**Comment by clearyandzap** [comment]
@justin808 Oh, I'm sorry. The safetyNet , here https://github.com/gaearon/react-hot-loader#hotcomponent-options

**Comment by csalvato** [comment]
I just used these instructions to set up my React app with HMR.

Why not just merge this as is? It can only help people.

**Comment by justin808** [comment]
@jakeNiemiec @gauravtiwari This PR is ready per requested changes.

**Comment by justin808** [comment]
@jakeNiemiec and @gauravtiwari I believe I've addressed all your comments.

**Comment by justin808** [comment]
@jakeNiemiec @gauravtiwari Any reason that this cannot be merged? It's just a doc improvement. I took great care to get this ready.

File: docs/react.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    inline: true
+```
+
+The basic setup will have HMR working with the default webpacker setup. However, the basic will cause a full page refresh each time you save a file.
+
+Webpack's HMR allows replacement of modules in-place without reloading the browser.
+
+To do this next part, you have two options:
+
+1. Use the
```
Comment (rchiba) [inline docs/react.md:80]: Is this line missing text?

Comment (justin808) [reply docs/react.md:80]: Yes -- fixing now. 😄

**Comment by rchiba** [comment]
For anyone who's using these docs / this thread to add HMR to their app, I had to do two things to get my Sass to be included as part of the HMR process:

1. In webpacker.yml, set `extract_css: false` for the development environment, which will cause styles to be compiled to a JavaScript module that webpack will insert as CSS into the page dynamically. [Additional documentation on extract_css.](https://rossta.net/blog/how-to-use-webpacker-yml.html#extract_css)
2. Doing step 1 caused my `stylesheet_pack_tag "application"` to throw an error in development as the styles are no longer being compiled into a separate stylesheet file, so I had to add a condition to prevent the error `stylesheet_pack_tag "application" if ENV["RAILS_ENV"] != "development"`

File: docs/react.md (hunk 1 above)
Comment (rchiba) [reply docs/react.md]: Thank you @justin808 for putting this documentation together, it was very helpful in getting HMR working for my app. 🙏

**Comment by justin808** [comment]
@rchiba you should not need step 2.

https://github.com/rails/webpacker/blob/d456b1b3783313bf0b3927bed6f434d1de5854c4/lib/webpacker/helper.rb#L127-L147

**Comment by justin808** [comment]
@rchiba see my last commit.

**Pull request merged**